File: Ejercicios Propuestos 2/Ejercicio 10/Ejercicio 10.py
NAME = "Renzo Leonardo Cienfuegos Cardenas"
GROUP = 2

# Bucle que se repite y muestra error si no es natural
while True:
    n = int(input("Introduzca un entero positivo: "))

    # Si se escribe un número menor a 1 sale el error
    if n < 1:
        print("¡Error! Debe ingresar un número natural.")
    else:
        break

factorial = 1
i = 1

while i <= n:
    factorial *= i
    i += 1

print(f"El factorial de {n} es: {factorial}")

File: Ejercicios Propuestos 2/Ejercicio 10/Ejercicio 10.jl
# Bucle que se repite y muestra error si no es natural
while true
    println("Introduzca un entero positivo: ")
    n = parse(Int, readline())
    
    # Si se escribe un número menor a 1 sale el error
    if n < 1
        println("¡Error! Debe ingresar un número natural.")
    else
        global n
        break # Hasta Que n >= 1
    end
end

factorial = 1
i = 1

# Bucle Mientras
while i <= n
    global factorial, i
    factorial = factorial * i
    i = i + 1
end

println("El factorial de ", n , " es: ", factorial)